### DISCUSSION TITLE
Graphql Query Error

### DISCUSSION THREAD

**Discussion by emmadigital** [gatsbyjs/gatsby#26997] [Help]
## Summary
I get graphql query error for featuredImage field, sometime it works fine but sometimes I get below error but when I run `gatsby clean` command 3 - 4 times the error goes away. I want to know the permanent solution to this.

## Relevant information
Here is the screenshot of the error and my graphql query for fetching featuredImage:

![image](https://user-images.githubusercontent.com/64609778/93942070-5d16e980-fd38-11ea-9044-98a74d5ac8a7.png)

```
 query TestimonialsPage($id: String!) {
    page: markdownRemark(id: { eq: $id }) {
      ...Meta
        html
        id
      frontmatter {
        title
        template
        date(formatString: "MMMM Do, YYYY")
        featuredImage
        excerpt
      }
    }
```
Screenshot of Graphql

![image](https://user-images.githubusercontent.com/64609778/93942459-1bd30980-fd39-11ea-85be-408d211be8c6.png)

My repro: https://github.com/emmadigital/austinlaw.git

My Netlify build sometime also fails because of this featuredImage issue. Please advise whats the permanent solution for this problem.

And there is no **publicURL** showing up in graphql which I can use instead.

**Comment by ascorbic** [comment]
Hi @emmadigital ,
The issue here is that `featuredImage` is being identified as a file, meaning it will be converted from a string into a `File` object. This is possibly down to gatsby-remark-relative-images, which looks for images in the frontmatter. That said, I can't reproduce this at all locally, so I can't say for sure, and I don't know why it's intermittent. As a start, can I suggest upgrading to the latest version of gatsby-remark-relative-images, and gatsby, and seeing if the problem persists. You also seem to have a few cases where a filename in the markdown has different casing from the file on disk. This can also cause issues, so fixing that may help.

**Comment by emmadigital** [comment]
> Hi @emmadigital ,
> The issue here is that `featuredImage` is being identified as a file, meaning it will be converted from a string into a `File` object. This is possibly down to gatsby-remark-relative-images, which looks for images in the frontmatter. That said, I can't reproduce this at all locally, so I can't say for sure, and I don't know why it's intermittent. As a start, can I suggest upgrading to the latest version of gatsby-remark-relative-images, and gatsby, and seeing if the problem persists. You also seem to have a few cases where a filename in the markdown has different casing from the file on disk. This can also cause issues, so fixing that may help.

Hi @ascorbic Thank you for reviewing my issue. Is this how we install latest version as listed below:
`npm install --save gatsby-remark-relative-images@1.1.1`
`npm install gatsby@latest`

_You also seem to have a few cases where a filename in the markdown has different casing from the file on disk._ Can you please share anysuch file so I can fix it.

**Comment by ascorbic** [comment]
@emmadigital I didn't do a comprehensive search, but the ones I saw at first were the pdfs listed in guides.md. 

The latest version of gatsby-remark-relative-images looks like it's 2.0.2. I should clarify that as it is a third party plugin, if this is where the issue lies you will need to raise a ticket there instead.

**Comment by emmadigital** [comment]
Hi @ascorbic I installed latest version of gatsby-remark-relative-images but no change I still get this error. I have to clean the cache again and again to get it successfully build. If you have any other solution please advise.

**Comment by emmadigital** [comment]
anyone can help me resolve this issue?

**Comment by wardpeet** [comment]
Sorry to hear you're running into an issue. To help us best begin debugging the underlying cause, it is incredibly helpful if you're able to create a minimal reproduction. This is a simplified example of the issue that makes it clear and obvious what the issue is and how we can begin to debug it.

If you're up for it, we'd very much appreciate if you could provide a minimal reproduction and we'll be able to take another look.

Thanks for using Gatsby! 💜

**Comment by emmadigital** [comment]
Thank you so much @wardpeet for your response. Here is my repo https://github.com/emmadigital/austinlaw.git
